test/integration/crcsuite/prepare.go

```diff
@@ -62,12 +63,43 @@ func ParseFlags() {
 
 	flag.Parse()
 	if flag.NArg() < 2 {
-		fmt.Printf("Invalid number of arguments, the paths to the bundle file and to the pull secret file are required\n")
+		fmt.Printf("Invalid number of arguments, the path to the pull secret file and the bundle version are required\n")
 		os.Exit(1)
 	}
-	bundleURL = flag.Args()[0]
-	_, bundleName = filepath.Split(bundleURL)
-	pullSecretFile = flag.Args()[1]
+
+	pullSecretFile = flag.Args()[0]
+
+	// embedded bundle
+	// this will never occur on Centos CI
+	if flag.NArg() == 2 {
+		bundleEmbedded = true
+		bundleVersion = flag.Args()[1]
+		// assume default hypervisor
+		var hypervisor string
+		switch os := runtime.GOOS; os {
+		case "darwin":
+			hypervisor = "hyperkit"
+		case "linux":
+			hypervisor = "libvirt"
+		case "windows":
+			hypervisor = "hyperv"
+		default:
+			fmt.Printf("Unsupported OS: %s", os)
+		}
+
+		bundleName = fmt.Sprintf("crc_%s_%s.crcbundle", hypervisor, bundleVersion)
+
+	}
+
+	// separate bundle
+	// 3: on laptop
+	// 4: on Centos CI
+	if flag.NArg() > 2 {
```

Comment:
Not sure I understand the 3: on laptop and 4: on Centos CI, we only make use of arg 1 and 2?

Reply:
I think we make use of arg 0 (pull secret) and arg 1 (bundle location). There's also arg 2 (bundle version) which is ignored except when bundle is embedded (and bundle location is not passed), in which case it's actually arg 1. For some reason, on Centos CI, there's a 4th argument which is always there (I forgot what it was). Luckily though, we don't care about the case when bundle is embedded on CentosCI, because we don't test that case there. So it can be clumped with `flag.NArg() >= 3` on a laptop (i.e. not on CentosCI). 

This was the best I could do if "embedded" was supposed to be specified by absence of `BUNDLE_LOCATION`. I agree it's really confusing and really flimsy hunk of code! Hence I decided to rework it: see my [binloc branch](https://github.com/jsliacan/crc/tree/binloc). The solution I had in mind for specifying the binary location doesn't work yet, so it's not there.